core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -315,6 +327,24 @@ private CompletableFuture<AggregatedHttpFile> loadSpecifications(Executor execut
             });
         }
 
+        private CompletableFuture<AggregatedHttpFile> loadSchemas(Executor executor,
+                                                                  ServiceSpecification spec) {
+            return files.computeIfAbsent(SCHEMAS_PATH, key -> {
+                return CompletableFuture.supplyAsync(() -> {
+                    try {
+                        final ArrayNode jsonSpec = JsonSchemaGenerator.generate(spec);
+
+                        final byte[] content = jsonMapper.writerWithDefaultPrettyPrinter()
+                                                         .writeValueAsBytes(jsonSpec);
+                        return toFile(content, MediaType.JSON_UTF_8);
+                    } catch (Exception e) {
+                        logger.warn("Failed to generate JSON schemas:", e);
+                        return toFile("[]".getBytes(), MediaType.JSON_UTF_8);
```

Comment:
How about throwing an exception as we do in the `loadSpecifications` method?

Reply:
I didn't want to block the whole application if something goes wrong in the JSON Schema generation. This behavior falls back to the old editor mode with no auto-complete feature. When the specification load is failed it basically makes the whole application useless (I suppose).

Reply:
That makes sense. 👍